### ISSUE TITLE
Regression: False positive - TypeVars aren't bound to 'Unknown' when non-subscripted type is passed to function

### ISSUE THREAD

**Issue by GugelRobin** [microsoft/pyright#5392]
**Describe the bug**
See code sample. TypeVars of `dict` aren't bound to `Unknown` leading to false positives.

**To Reproduce**
```python
from typing import Any, TypeVar, reveal_type

T = TypeVar("T")

def my_cast(t: type[T], val: Any) -> T:
    return val


x = my_cast(dict, {"foo": "bar"})

reveal_type(x) # Type of "x" is "dict[_KT@dict, _VT@dict]"
print(x["foo"]) # Argument of type "Literal['foo']" cannot be assigned to parameter "__key" of type "_KT@dict" in function "__getitem__"
```

**Expected behavior**
`reveal_type(x)` should be `dict[Unknown, Unknown]` and `print(x["foo"])` should have any errors

**VS Code extension or command-line**
pyright command line version 1.1.316

**Additional Information**
Using `typing.cast` doesn't result in that problem. In my actual code there're reasons though why I don't wanna use `typing.cast`.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.318, which I just published. It will also be included in a future release of pylance.
